crosshair/libimpl/collectionslib.py

```diff
@@ -8,14 +8,71 @@ class ListBasedDeque:
     def __init__(self, contents: List[T], maxlen: Optional[int]=None):
```

Comment:
Making the class extend `collections.abc.Sequence` is possibly a good idea?
Which makes me also think that some list-like deque behavior might not work properly yet?:
```
>>> from collections import deque
>>> 3 in deque([3,4])
True
>>> deque([3,4]) < deque([4,3])
True
>>> deque([3,4]) > deque([4,3])
False
>>> deque([3,4]) == deque([3,4])
True
```